src/libmongoc/doc/mongoc_uri_t.rst

```diff
@@ -106,6 +106,13 @@ MONGOC_URI_SRVMAXHOSTS                     srvmaxhosts                       0
 
 Setting any of the \*timeoutMS options above to ``0`` will be interpreted as "use the default value".
 
+.. warning::
+
+  Setting any of the \*timeoutMS options above to either ``0`` or a negative value is discouraged due to unspecified and inconsistent behavior.
```

Comment:
Not sure how to reconcile the documentation above:

> Setting any of the \*timeoutMS options above to ``0`` will be interpreted as "use the default value".

With the documentation/use of timeout parameters:

> The number of milliseconds to wait before failure, a timeout of 0 will not block.

Therefore, included value of `0` in the new warning paragraph.

Reply:
I suggest removing:
> Setting any of the \*timeoutMS options above to ``0`` will be interpreted as "use the default value".

I expect the actual behavior does not match. Setting `socketTimeoutMS=0` passes 0 for the timeout to `readv` and `writev`, not the value for `MONGOC_DEFAULT_SOCKETTIMEOUTMS`.

Reply:
Removed as requested.